src/components/wallet/use-hiiiq-balance.ts

```diff
@@ -56,17 +56,13 @@ export const useHiIQBalance = (address: string | undefined | null) => {
   useEffect(() => {
     const getBalance = async () => {
       const hiiqBalance = Number(
-        formatUnits(balanceOf as unknown as bigint, 18),
+        formatUnits(BigInt(balanceOf?.toString() ?? 0), 18),
```

Comment:
Incase balanceOf is bigint or null, we can do type checking so it reads less complex

```ts
const fetchedBalance = typeof balanceOf === "bigint" ? balanceOf : 0
const hiiqBalance = Number(formatUnits(fetchedBalance), 18)
```

Reply:
its not of type bigint, its Result | never but i will make it less complex

Reply:
Ohh, then cool !